apps/documents/models.py

```diff
@@ -1,16 +1,47 @@
-from django.db import models
+from django.db import models, transaction
+from django.urls import reverse
+from field_audit import audit_fields
+from field_audit.models import AuditingManager
 
-from apps.pipelines.models import Node
+from apps.experiments.versioning import VersionDetails, VersionField, VersionsMixin, VersionsObjectManagerMixin
 from apps.teams.models import BaseTeamModel
 from apps.utils.conversions import bytes_to_megabytes
+from apps.utils.deletion import get_related_pipelines_queryset
 
 
-class Collection(BaseTeamModel):
+class CollectionObjectManager(VersionsObjectManagerMixin, AuditingManager):
+    pass
+
+
+@audit_fields(
+    "name",
+    audit_special_queryset_writes=True,
+)
+class Collection(BaseTeamModel, VersionsMixin):
     name = models.CharField(max_length=255)
     files = models.ManyToManyField("files.File", blank=False)
+    working_version = models.ForeignKey(
+        "self",
+        on_delete=models.CASCADE,
+        null=True,
+        blank=True,
+        related_name="versions",
+    )
+    is_archived = models.BooleanField(default=False)
+    version_number = models.PositiveIntegerField(default=1)
+
+    objects = CollectionObjectManager()
 
     class Meta:
-        constraints = [models.UniqueConstraint(fields=["team", "name"], name="unique_collection_per_team")]
+        constraints = [
+            models.UniqueConstraint(
+                fields=["team", "name", "version_number", "working_version_id"],
+                name="unique_collection_version_per_team",
+            )
+        ]
+
+    def __str__(self) -> str:
+        return f"{self.name}"
```

Comment:
f-string not required:

```suggestion
        return self.name
```